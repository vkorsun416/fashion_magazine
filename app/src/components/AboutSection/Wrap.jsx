import React from 'react';
import css from "./Wrap.module.css";

function Wrap(props) {
    const {children} = props;

    return (
        <div className={css.wrapper}>
            <div className={css.container}>
                <div className={css.grid}>
                    {children}
                </div>
            </div>
        </div>
    );
}

export default Wrap;
