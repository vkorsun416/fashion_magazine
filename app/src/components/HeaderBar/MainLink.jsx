import React from "react";
import {Link, Route, Switch} from "react-router-dom";
import css from './MainLink.module.css';
import {LIGHT} from "../../layout/themes";

function MainLink(props) {
    const {theme} = props;

    return (
        <Switch>
            <Route
                path={"/"}
                strict
                exact
            >
                <div className={css.text + " " + (theme === LIGHT ? css.light : "")}>
                    INTERVIEW
                </div>
            </Route>

            <Route>
                <Link
                    to={"/"}
                    className={css.link}
                >
                    <div className={css.texts + " " + (theme === LIGHT ? css.light : "")}>
                        INTERVIEW
                    </div>
                </Link>
            </Route>
        </Switch>
    );
}

export default MainLink;
