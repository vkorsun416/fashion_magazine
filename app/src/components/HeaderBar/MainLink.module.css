.link {
    text-decoration: none;
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
}

.text,
.texts {
    position: relative;
    font-size: 18px;
    z-index: 2;
    width: 92px;
    height: 12px;
    color: #fff;
    padding: 24px;
    left: 0;
}

.texts:before {
    position: absolute;
    content: "";
    display: block;
    top: 100%;
    left: 0;
    width: 0;
    height: 1px;
    background-color: #fff;
    transition: all 0.4s ease-out;
}

.texts:hover:before {
    right: auto;
    width: 100%;
}

.light {
    color: #000;
}

.light:before {
    background-color: #000;
}
