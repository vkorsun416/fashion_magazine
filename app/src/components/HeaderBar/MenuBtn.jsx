import React from "react";
import css from "./MenuBtn.module.css";
import {LIGHT} from "../../layout/themes";

function MenuBtn(props) {
    const {openMenu, theme} = props;

    return (
        <div
            className={css.btn + " " + (theme === LIGHT ? css.dark : "")}
            onClick={openMenu}
        >
            <div className={css.item + " " + (theme === LIGHT ? css.dark : "")}/>
        </div>
    );
}

export default MenuBtn;
