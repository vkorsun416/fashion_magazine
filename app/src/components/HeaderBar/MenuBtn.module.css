.btn {
    display: flex;
    align-items: center;
    width: 40px;
    height: 21px;
    cursor: pointer;
    padding: 24px;
    right: 0;
    position: relative;
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
}

.item {
    width: 40px;
    height: 3px;
    background-color: #fff;
}

.btn::before, .btn::after {
    content: "";
    width: 40px;
    height: 3px;
    background-color: #fff;
    position: absolute;
}

.btn.dark::before, .btn.dark::after {
    background-color: #000;
}

.btn::before {
    top: 24px;
}

.btn::after {
    bottom: 24px;
}

.item {

}

.item.dark {
    background-color: #000;
}

.btn.active .item,
.btn.active::before,
.btn.active::after {
    background-color: #000;
    opacity: 1;
}

.btn.active .item {
    opacity: 0;
}

.btn.active::before {
    transform: translateY(10px) rotate(-45deg);
}

.btn.active::after {
    transform: translateY(-7px) rotate(45deg);
}
