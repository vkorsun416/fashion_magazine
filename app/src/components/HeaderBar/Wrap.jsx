import React, {Component} from 'react';
import css from "./Wrap.module.css";

class Wrap extends Component {
    state = {
        isScrolled: false,
    }

    componentDidMount() {
        window.addEventListener('scroll', () => {
            const isScrolled = window.pageYOffset > document.documentElement.clientHeight;

            if (isScrolled !== this.state.isScrolled) {
                this.setState({isScrolled})
            }
        });
    }

    /**
     *
     * Возвращает фон для хедера,
     * если скролл выше высоты экрана добавляем фон
     */
    getClasses() {
        const {isScrolled} = this.state;
        const classes = [css.rgba];

        if (isScrolled) {
            classes.push(css.blur);
        }

        return classes;
    }

    renderWrap() {
        const {children} = this.props;

        return (
            <header>
                <div className={this.getClasses().join(" ")}>
                    <div className={css.container}>
                        <div className={css.header}>
                            {children}
                        </div>
                    </div>
                </div>
            </header>
        )
    }

    render() {
        return (
            this.renderWrap()
        );
    }
}

export default Wrap;
