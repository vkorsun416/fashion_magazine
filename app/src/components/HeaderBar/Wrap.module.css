.rgba {
    position: fixed;
    width: 100%;
    top: 0;
    left: 0;
    right: 0;
    z-index: 4;
    transition: all 800ms ease-in-out;
}

.blur {
    background: rgba(107, 112, 75, 0.44);
    backdrop-filter: blur(5px);
    transition: all 800ms ease-in-out;
}

.container {
    padding: 0;
    max-width: 1440px;
    margin: 0 auto;
}

.header {
    display: flex;
    justify-content: space-between;
    align-items: center;
}