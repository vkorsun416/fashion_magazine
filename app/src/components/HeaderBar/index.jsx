import React from "react";
import PropTypes from "prop-types";
import MainLink from "./MainLink";
import MenuBtn from "./MenuBtn";
import Wrap from "./Wrap";
import * as themes from "../../layout/themes";
import {withPopup} from "../../context/PopupContext/withPopup";

function HeaderBar(props) {
    const {openMenu, theme} = props;

    return (
        <Wrap>
            <MainLink theme={theme}/>

            <MenuBtn
                theme={theme}
                openMenu={openMenu}
            />
        </Wrap>
    );
}

HeaderBar.propTypes = {
    theme: PropTypes.oneOf(Object.values(themes)),
    openMenu: PropTypes.func.isRequired,
};

export default withPopup(HeaderBar);
