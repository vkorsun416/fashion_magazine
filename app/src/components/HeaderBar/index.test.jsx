import React from "react";
import {mount} from "enzyme";
import HeaderBar from ".";
import {StaticRouter} from "react-router-dom";
import Wrap from "./Wrap";
import MainLink from "./MainLink";
import MenuBtn from "./MenuBtn";
import {LIGHT} from "../../layout/themes";

globalThis.console.error = jest.fn();

describe("MainHeaderBar bar", function () {
    test("Should render to string", () => {
        const wrap = mount(
            <StaticRouter>
                <HeaderBar />
            </StaticRouter>
        );

        expect(wrap.html()).toMatchSnapshot();
    });

    test("Should receive prop from context", () => {
        const wrap = mount(
            <StaticRouter>
                <HeaderBar />
            </StaticRouter>
        );

        expect(wrap.find(MenuBtn).props().openMenu).toBe(undefined);
    });

    test("Should render with dependencies", () => {
        const wrap = mount(
            <StaticRouter>
                <HeaderBar />
            </StaticRouter>
        );

        expect(wrap.find(Wrap)).toHaveLength(1);
        expect(wrap.find(Wrap).props().children).toHaveLength(2);

        expect(wrap.find(MainLink)).toHaveLength(1);
        expect(wrap.find(MainLink).props().theme).toBe(undefined);

        expect(wrap.find(MenuBtn)).toHaveLength(1);
        expect(wrap.find(MenuBtn).props().theme).toBe(undefined);
    });

    test("Should render with theme", () => {
        const wrap = mount(
            <StaticRouter>
                <HeaderBar
                    theme={LIGHT}
                />
            </StaticRouter>
        );

        expect(wrap.find(MainLink).props().theme).toBe(LIGHT);
        expect(wrap.find(MenuBtn).props().theme).toBe(LIGHT);
        expect(wrap.html()).toMatchSnapshot();
    });

    test("Should render with wrong theme", () => {
        globalThis.console.error.mockClear();

        mount(
            <StaticRouter>
                <HeaderBar
                    theme={"WRONG"}
                />
            </StaticRouter>
        );

        expect(globalThis.console.error).toBeCalledTimes(1);
    });
});

describe("Main link", () => {
    test("Should render on main", () => {
        const wrap = mount(
            <StaticRouter location={"/"}>
                <HeaderBar />
            </StaticRouter>
        );

        expect(wrap.find(MainLink).html()).toMatchSnapshot();
    });

    test("Should render other page", () => {
        const wrap = mount(
            <StaticRouter location={"/page"}>
                <HeaderBar />
            </StaticRouter>
        );

        expect(wrap.find(MainLink).html()).toMatchSnapshot();
    });
});

describe("MenuButton", () => {
    test("Should render with popup", () => {
        const openMenu = jest.fn();

        const wrap = mount(
            <StaticRouter location={"/"}>
                <HeaderBar
                    openMenu={openMenu}
                />
            </StaticRouter>
        );

        wrap.find(".btn").simulate("click");

        expect(openMenu).toBeCalledTimes(1);
    });
});
