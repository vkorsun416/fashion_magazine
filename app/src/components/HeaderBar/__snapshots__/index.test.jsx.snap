// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header bar Should render to string 1`] = `"<header><div class=\\"rgba\\"><div class=\\"container\\"><div class=\\"header\\"><div class=\\"text \\">INTERVIEW</div><div class=\\"btn \\"><div class=\\"item \\"></div></div></div></div></div></header>"`;

exports[`Header bar Should render with theme 1`] = `"<header><div class=\\"rgba\\"><div class=\\"container\\"><div class=\\"header\\"><div class=\\"text light\\">INTERVIEW</div><div class=\\"btn dark\\"><div class=\\"item dark\\"></div></div></div></div></div></header>"`;

exports[`Main link Should render on main 1`] = `"<div class=\\"text \\">INTERVIEW</div>"`;

exports[`Main link Should render other page 1`] = `"<a class=\\"link\\" href=\\"/\\"><div class=\\"texts \\">INTERVIEW</div></a>"`;
